typeid/typeid-go/constructors.go

```diff
@@ -98,22 +98,39 @@ func split(id string) (string, string, error) {
 	}
 }
 
-// FromUUID encodes the given UUID (in hex string form) as a TypeID with the given prefix.
-func FromUUID[T Subtype, PT SubtypePtr[T]](prefix string, uidStr string) (T, error) {
+// FromUUID encodes the given UUID (in hex string form) as a TypeID
+func FromUUID[T Subtype, PT SubtypePtr[T]](uidStr string) (T, error) {
 	uid, err := uuid.FromString(uidStr)
 	var nilID T
 
 	if err != nil {
 		return nilID, err
 	}
 	suffix := base32.Encode(uid)
-	return from[T, PT](prefix, suffix)
+	return from[T, PT](nilID.Prefix(), suffix)
+}
+
+// FromUUIDBytes encodes the given UUID (in byte form) as a TypeID
+func FromUUIDBytes[T Subtype, PT SubtypePtr[T]](bytes []byte) (T, error) {
+	uidStr := uuid.FromBytesOrNil(bytes).String()
+	return FromUUID[T, PT](uidStr)
+}
+
+// AnyPrefixFromUUID encodes the given UUID (in hex string form) as a TypeID with the given prefix.
+func AnyPrefixFromUUID(prefix string, uidStr string) (AnyID, error) {
+	uid, err := uuid.FromString(uidStr)
+
+	if err != nil {
+		return AnyID{}, err
+	}
+	suffix := base32.Encode(uid)
+	return from[AnyID, *AnyID](prefix, suffix)
 }
 
 // FromUUID encodes the given UUID (in byte form) as a TypeID with the given prefix.
-func FromUUIDBytes[T Subtype, PT SubtypePtr[T]](prefix string, bytes []byte) (T, error) {
+func AnyPrefixFromUUIDBytes(prefix string, bytes []byte) (AnyID, error) {
```

Comment:
Could we do: `FromUUIDBytesWithPrefix` and `FromUUIDWithPrefix` (matches the `WithPrefix` constructor we already have)